### PR TITLE
Move apis and client under kubeedge/cloud/pkg/ to kubeedge/pkg/

### LINKED ISSUE #3669: Move apis under kubeedge/cloud/pkg/apis to kubeedge/pkg/apis
Let's move all the apis to one place.

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:
Move apis and client under kubeedge/cloud/pkg/ to kubeedge/pkg/
**Which issue(s) this PR fixes**:

Fixes #3669 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/hack/verify-codegen.sh [hunk 1]
```diff
@@ -19,10 +19,11 @@ set -o nounset
...
 
 SCRIPT_ROOT=$(dirname "${BASH_SOURCE}")/..
+KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 
-DIFFROOT="${SCRIPT_ROOT}/pkg"
-TMP_DIFFROOT="${SCRIPT_ROOT}/_tmp/pkg"
-_tmp="${SCRIPT_ROOT}/_tmp"
+DIFFROOT="${KUBEEDGE_ROOT}/pkg"
+TMP_DIFFROOT="${KUBEEDGE_ROOT}/_tmp/pkg"
+_tmp="${KUBEEDGE_ROOT}/_tmp"
```
Comment (fisherxu) [inline cloud/hack/verify-codegen.sh:26]: The scripts can also be moved to root hack?

Comment (gy95) [reply cloud/hack/verify-codegen.sh:26]: yes, i guess that all the contents behind dir `cloud/hack` need moving to the root hack. And IMHO, it's better to hold those in one separate dir, and which dir name is better? codegen?

Comment (fisherxu) [reply cloud/hack/verify-codegen.sh:26]: The `cloud/hack/test-integration.sh` seems can be deleted. `cloud/hack/boilerplate` can move to hack, add add a dir `tools` to place the tool.go, other scripts can place under hack directly.

Comment (gy95) [reply cloud/hack/verify-codegen.sh:25]: fixed

**Review by fisherxu — COMMENTED** [review]
/approve

**Review by fisherxu — COMMENTED** [review]
/lgtm

**Pull request merged**